src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -392,7 +398,29 @@ public static void SetWindowPosition(int left, int top)
 
         public static void SetWindowSize(int width, int height)
         {
-            throw new PlatformNotSupportedException();
+            if (Console.IsOutputRedirected)
+                throw new IOException(SR.InvalidOperation_SetWindowSize);
+            if (width <= 0)
+                throw new ArgumentOutOfRangeException(nameof(width), width, SR.ArgumentOutOfRange_NeedPosNum);
+            if (height <= 0)
+                throw new ArgumentOutOfRangeException(nameof(height), height, SR.ArgumentOutOfRange_NeedPosNum);
+
+           lock (Console.Out)
+           {
+               Interop.Sys.WinSize winsize = default;
+               winsize.Row = (ushort)height;
+               winsize.Col = (ushort)width;
+               if (Interop.Sys.SetWindowSize(in winsize) == 0)
+               {
+                   s_windowWidth = winsize.Col;
+                   s_windowHeight = winsize.Row;
+               }
+               else
+               {
+                   Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
+                   throw Interop.GetExceptionForIoErrno(errorInfo);
```

Comment:
This should continue to throw a PlatformNotSupportedException if it's failing because Interop.Sys.SetWindowSize isn't implemented on the target platform.

Reply:
Fixed. Thank you.